src/v2i-hub/PedestrianPlugin/src/FLIRWebSockAsyncClnSession.cpp

```diff
@@ -276,6 +276,7 @@ namespace PedestrianPlugin
 
 		                std::lock_guard<mutex> lock(_psmLock);
                         psmxml = psm_xml_str;
+                        psmQueue.push(psmxml);
```

Comment:
I think you should add a mutex lock here also if you make the change I suggested below.

Reply:
An additional lock to the one two lines above?

Reply:
No, I missed that you had one already.